extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -311,25 +319,16 @@ class QueueProcessor extends EventEmitter {
                 task = new EchoBucket(this);
             }
             // ignore bucket entry if echo mode disabled
-        } else if (sourceEntry instanceof ObjectQueueEntry) {
-            const replicationStorageClass =
-                sourceEntry.getReplicationStorageClass();
-            const sites = replicationStorageClass.split(',');
-            return async.each(sites, (site, cb) => {
-                const replicationEndpoint = this.destConfig.bootstrapList
-                    .find(endpoint => endpoint.site === site);
-                if (replicationEndpoint
-                    && replicationBackends.includes(replicationEndpoint.type)) {
-                    task = new MultipleBackendTask(this, site);
-                } else {
-                    task = new ReplicateObject(this, site);
-                }
-                this.logger.debug('source entry is being pushed',
-                  { entry: sourceEntry.getLogInfo() });
-                return this.taskScheduler.push({ task, entry: sourceEntry },
-                                               sourceEntry.getCanonicalKey(),
-                                               cb);
-            }, err => done(err));
+        } else if (sourceEntry instanceof ObjectQueueEntry &&
+        sourceEntry.getReplicationStorageClass().includes(this.site)) {
```

Comment:
You will need to keep the conversion to an array first, to avoid side-effects of substring matching (e.g. `cloud` location would also match `cloud1` with string matching)